distributed-commandbus-jgroups/src/main/java/org/axonframework/jgroups/commandhandling/JoinMessage.java

```diff
@@ -74,15 +79,24 @@ public int getLoadFactor() {
 
     @Override
     public void writeExternal(ObjectOutput out) throws IOException {
-        out.writeObject(address);
+
+        try {
+            Util.writeAddress(address, out);
+        } catch (Exception e) {
+            logger.error("Serialize exception: {}", e);
+        }
         out.writeInt(loadFactor);
         out.writeObject(messageFilter);
     }
 
     @SuppressWarnings("unchecked")
     @Override
     public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
-        address = (Address) in.readObject();
+        try {
+            address = Util.readAddress(in);
+        } catch (Exception e) {
+            logger.error("Serialize exception: {}", e);
```

Comment:
Could you add `address` to that log error. Something like `Exception while serializing address: {}`

Reply:
good idea.